components/VM_Arm/src/main.c

```diff
@@ -633,7 +656,17 @@ int install_vm_devices(vm_t *vm)
     /* Install virtual devices */
     if (config_set(CONFIG_VM_PCI_SUPPORT)) {
         err = vm_install_vpci(vm, io_ports, pci);
-        assert(!err);
+        if (err) {
+            ZF_LOGE("Failed to install VPCI device (%d)", err);
+            return -1;
+        }
+    }
+
+    /* Install and initialize guest RAM. */
+    err = init_ram_module(vm);
```

Comment:
As explained above, `cookie` parameter is missing. I think it is enough to pass NULL here.